### PR TITLE
ci: add docs github actions

### PR DESCRIPTION
**Pull request by wischli**
Adds Github actions to build and publish docs. Tested locally beforehand: https://wischli.github.io/runtime-pallet-library/mock_builder/index.html

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Rustdoc
+on:
+  push:
+    branches: [main]
+  pull_request:
```
Comment (lemunozm) [inline .github/workflows/docs.yml:5]: Not having the `main` branch here means that if some doc fails, we will know once it's merged in `main`? Reformulating the question, do we want to check the doc integrity for each PR before merging it?

**Review by lemunozm — APPROVED** [review]
Awesome! Thanks for tacking this!! 🙌🏻

File: .github/workflows/docs.yml (hunk 1 above)
Comment (wischli) [reply .github/workflows/docs.yml:5]: Good point. We should know it would fail before merging for sure!

Comment (wischli) [reply .github/workflows/docs.yml:5]: This CI actually builds the docs on PRs against `main`, it just doesn't publish: https://github.com/foss3/runtime-pallet-library/actions/runs/6550910806/job/17790998211?pr=4

Comment (lemunozm) [reply .github/workflows/docs.yml:5]: Right! thanks

**Pull request merged**